src/utils/mod.rs

```diff
@@ -157,6 +160,30 @@ pub fn get_user_from_request(req: &HttpRequest) -> Result<String, RBACError> {
     Ok(user_id)
 }
 
+pub async fn delete_correlations_filters_dashboards_for_user(
+    req: &HttpRequest,
+) -> Result<(), RBACError> {
+    let session_key = extract_session_key_from_req(req)
+        .map_err(|err| RBACError::Anyhow(anyhow::Error::msg(err.to_string())))?;
+
+    let user_id = get_user_from_request(req).map(|s| get_hash(&s.to_string()))?;
+    CORRELATIONS
+        .delete_correlations_for_user(&user_id, &session_key)
+        .await
+        .map_err(|err| RBACError::Anyhow(anyhow::Error::msg(err.to_string())))?;
+
+    let dashboards = DASHBOARDS.list_dashboards_by_user(&user_id);
+    for dash in dashboards.iter() {
+        DASHBOARDS.delete_dashboard(dash.dashboard_id.as_ref().unwrap());
+    }
```

Comment:
```suggestion
    DASHBOARDS.delete_user(user_id)
```
This should be simplified further as so should the mutation of filter deletion